### PR TITLE
FIXED: Text overflows in the user management card details #9532

### LINKED ISSUE #9532: Text is overlapping on the user management page
**Describe the bug**
Text overflows in the user management card details, and text overlaps on mobile screens

**To Reproduce**

1. Go to 'Users on sidebar'
2. switch to mobile size
3. See error

**Expected behavior**
The text within the user management card should fit properly inside the container without overflowing and overlapping. On mobile screens, the text should wrap appropriately to maintain a responsive and visually consistent layout.

**Screenshots**

problem 1
Image

problem 2
Image

problem 3
Image

**Desktop (please complete the following information):**

- OS: [windows]
- Browser [chrome]
- Version [131.0.6778.205]

### PR DESCRIPTION
**Pull request by piyushyadav0191**
## Proposed Changes

- Fixes #9532 
- Added responsive max-width constraints for different screen sizes (sm/md/lg)
- Improved text truncation with proper ellipsis for long names
- Added tooltips to show full text on hover
- Fixed mobile layout with proper wrapping

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced tooltip functionality for enhanced user interaction in the user list and card display.
	- User's name now features tooltips that provide additional information on hover.

- **Style**
	- Adjusted layout for improved spacing and alignment of user information.

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
let's only do for name and not username. username already has validations to disallow long usernames

**Comment by piyushyadav0191** [comment]
Okay, let me take care of it.

**Comment by piyushyadav0191** [comment]
@rithviknishad Fixed what you asked for

File: src/components/Users/UserListAndCard.tsx [hunk 1]
```diff
@@ -77,30 +84,51 @@ const getNameAndStatusCard = (
...
-          </span>
+
+          <div className="min-w-0 max-w-full sm:max-w-[200px] md:max-w-[300px] lg:max-w-[400px]">
+            <span
+              className="text-sm text-gray-500 block text-ellipsis overflow-hidden whitespace-nowrap"
+              id={`username-${user.username}`}
+            >
+              {user.username}
+            </span>
+          </div>
```
Comment (rithviknishad) [inline src/components/Users/UserListAndCard.tsx]: do revert all changes to the username. there are still some changes

Comment (piyushyadav0191) [reply src/components/Users/UserListAndCard.tsx]: You mean I should revert all changes to username means username inside that tooltip or the original code?

Comment (rithviknishad) [reply src/components/Users/UserListAndCard.tsx]: keep original code

Comment (piyushyadav0191) [reply src/components/Users/UserListAndCard.tsx]: Okay, I will just use original code and add tooltip to name only as other maintainer suggested

**Comment by piyushyadav0191** [comment]
@rithviknishad created PR of what you asked for

**Review by rithviknishad — APPROVED** [review]

File: src/components/Users/UserListAndCard.tsx [hunk 2]
```diff
@@ -81,9 +88,21 @@ const getNameAndStatusCard = (
...
-              {formatName(user)}
-            </h1>
+            <TooltipProvider delayDuration={200}>
+              <Tooltip>
+                <TooltipTrigger asChild>
+                  <h1
+                    id={`name-${user.username}`}
+                    className="text-base font-bold"
+                  >
+                    {formatName(user)}
```
Comment (Jacobjeevan) [inline src/components/Users/UserListAndCard.tsx:99]: image

The tooltip works, however the long name is still displayed; issue isn't solved.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Users/UserListAndCard.tsx (hunk 2 above)
Comment (piyushyadav0191) [reply src/components/Users/UserListAndCard.tsx:99]: Either, I can truncate or can decide some width or height to that popover

Comment (Jacobjeevan) [reply src/components/Users/UserListAndCard.tsx:99]: > Either, I can truncate or can decide some width or height to that popover

Truncate the name on the card (Tooltip trigger). Tooltip Content is fine.

Comment (piyushyadav0191) [reply src/components/Users/UserListAndCard.tsx:99]: done

File: src/components/Users/UserListAndCard.tsx [hunk 3]
```diff
@@ -93,7 +93,7 @@ const getNameAndStatusCard = (
                 <TooltipTrigger asChild>
                   <h1
                     id={`name-${user.username}`}
-                    className="text-base font-bold"
+                    className="text-base font-bold truncate max-w-[200px]"
```
Comment (rithviknishad) [inline src/components/Users/UserListAndCard.tsx]: use rem values over pixels. pixels can behave differently on different DPIs

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/components/Users/UserListAndCard.tsx (hunk 3 above)
Comment (piyushyadav0191) [reply src/components/Users/UserListAndCard.tsx]: done

**Comment by rithviknishad** [comment]
image

**Comment by piyushyadav0191** [comment]
@rithviknishad if there is no extra changes required !!Can you merge the PR?

File: src/components/Users/UserListAndCard.tsx [hunk 4]
```diff
@@ -81,9 +88,21 @@ const getNameAndStatusCard = (
...
+              <Tooltip>
+                <TooltipTrigger asChild>
+                  <h1
+                    id={`name-${user.username}`}
+                    className="text-base font-bold truncate max-w-[7rem]"
+                  >
+                    {formatName(user)}
+                  </h1>
+                </TooltipTrigger>
+                <TooltipContent>
```
Comment (Jacobjeevan) [inline src/components/Users/UserListAndCard.tsx]: I'd suggest a bit more styling to the tooltip content. Check the tooltip for Occupancy Badge in Facility for instance:

image

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Users/UserListAndCard.tsx (hunk 4 above)
Comment (piyushyadav0191) [reply src/components/Users/UserListAndCard.tsx]: Okay, I just added more styling and dynamic transitions on state and side.  @Jacobjeevan

File: src/components/Users/UserListAndCard.tsx [hunk 5]
```diff
@@ -81,9 +89,24 @@ const getNameAndStatusCard = (
...
+                <TooltipTrigger asChild>
+                  <h1
+                    id={`name-${user.username}`}
+                    className="text-base font-bold truncate max-w-[7rem] inline-block"
+                  >
+                    {formatName(user)}
+                  </h1>
+                </TooltipTrigger>
+                <TooltipContent
+                  className="px-4 py-3 text-sm bg-black text-white rounded-xl shadow-xl border border-indigo-300 animate-in fade-in-0 zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2"
```
Comment (Jacobjeevan) [inline src/components/Users/UserListAndCard.tsx]: Why add any of this? Shadcn already handles the animation, let's not modify that.

```suggestion
                  className="px-4 py-3 text-sm bg-black text-white rounded-md shadow-lg"
```

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**